auth/jwt.go

```diff
@@ -38,6 +46,25 @@ type JWTConfigCommon struct {
 	ChannelsClaim string `json:"channels_claim"`
 }
 
+// ValidFor returns whether the issuer matches, and one of the audiences matches
+func (j JWTConfigCommon) ValidFor(issuer string, audiences audience) bool {
+	if j.Issuer != issuer {
+		return false
+	}
+	// Validation enforces that ClientID is non-nil for OIDC.
+	if j.ClientID == nil || *j.ClientID == "" {
+		return true
```

Comment:
The updated comment still sounds like j.ClientID==nil should return false, while j.ClientID="" should return true.  I think this is more than a nitpick, as we want to avoid bugs that fail to set j.ClientID and result in audience validation being skipped.